R/attributes.R

```diff
@@ -539,28 +539,36 @@ vertex.attributes <- function(graph, index = V(graph)) {
     any(names(value) == "") || any(duplicated(names(value)))) {
     stop("Value must be a named list with unique names")
   }
-  if (any(sapply(value, length) != length(index))) {
+  if (!all(sapply(value, length) == length(index))) {
     stop("Invalid attribute value length, must match number of vertices")
   }
 
   if (!missing(index)) {
     index <- as_igraph_vs(graph, index)
 
-    if (any(duplicated(index)) || any(is.na(index))) {
+    if ((anyDuplicated(index) > 0) || (anyNA(index) > 0)) {
       stop("Invalid vertices in index")
     }
   }
 
+  index_number_not_vcount <- (length(index) != vcount(graph))
+  index_not_vertices_indices <- (!all(index %in% V(graph)))
   if (!missing(index) &&
-    (length(index) != vcount(graph) || any(index != V(graph)))) {
-    vs <- V(graph)
```

Comment:
no need to create this intermediary value as one can use `vcount(graph)` directly.